### DISCUSSION TITLE
How to handle multiple fields in a custom component?

### DISCUSSION THREAD

**Discussion by wuarmin** [react-hook-form/react-hook-form#8350] [Q&A] [answered]
Hey,

this is just a simple pseudo-example, but I think you should see what's my goal. 

```javascript
const { control, register, setValue } = useForm({
  defaultValues: { from: 2010, to: 2020 }
});

return (
    <>
      <InlineInput label="From">
        <Controller
          control={control}
          name="from"
          render={({ field }) => (
            <YearPicker value={field.value} onChange={field.onChange} />
          )}
        />
      </InlineInput>
      <InlineInput label="To">
        <Controller
          control={control}
          name="to"
          render={({ field }) => (
           <YearPicker value={field.value} onChange={field.onChange} />
          )}
        />
      </InlineInput>
    </>
);

```
I want to refactor it to the following. Can you show me how to handle it with react-hook-form the properly?

```javascript
const { control, register } = useForm({
  defaultValues: { from: 2010, to: 2020, name: "test" }
});

return (
    <>
      <InlineInput label="From">
        <Controller
          control={control}
          name="??" // how to handle multiple fields in a custom component?
          render={({ field }) => ( //
            <FromToYearPicker from={??} to={??} onChange={({field, value}) => { setValue(field, value) }} /> //
          )}
        />
      </InlineInput>
    </>
);

```

Thank you and best regards

**Comment by wuarmin** [comment, marked as answer]
I solved it as follows:

```javascript
const { control, register, watch, setValue } = useForm({
  defaultValues: { from: 2010, to: 2020, name: "test" }
});

return (
    <>
      <FromToYearPicker
        from={watch("from")}
        to={watch("to")}
        onChange={setValue}
      />
    </>
);
```
Is that the right  way to solve such a scenario?

**Reply by bluebill1049** [reply]
you should just use `useController` instead use `watch` API, this could a be single input with nested object data.

```
defaultValues: { test: { from: 2010, to: 2020 } , name: "test" }
useControlelr({
  name: 'test'
})
```

**Reply by wuarmin** [reply]
Thanks, but I cannot handle it with a nested `from` and `to`. In this case the props has to be on the first level. WDYT? What's the disadvantage to handle it with `watch`?

**Reply by bluebill1049** [reply]
> Thanks, but I cannot handle it with a nested from and to. In this case the props has to be on the first level. WDYT? What's the disadvantage to handle it with watch?

I am not sure what do you mean above? watch is meant to used before render and subscribed to fields update.

**Reply by wuarmin** [reply]
@bluebill1049 sorry for my late reply. I oversaw this. The issue is, that the result object needs `from` and `to` at the first level and not nested in under `test`-key. A transformation, afterwards, from `{ name: "test", test: { from: "1989-06-06", to: "1990-06-06" }` to `{ name: "test", from: "1989-06-06", to: "1990-06-06" }` has to be avoided, so I need to find a solution to handle two fields (`from` and `to`) in one `controller`.
